service/src/builder.rs

```diff
@@ -291,22 +182,107 @@ fn ensure_binary(package: &Package) -> anyhow::Result<()> {
 
 /// Make sure "cdylib" is set for shuttle-next projects, else set it if possible.
 fn ensure_cdylib(package: &Package) -> anyhow::Result<()> {
-    if package.targets().iter().any(|target| target.is_lib()) {
+    if package.targets.iter().any(is_cdylib) {
         Ok(())
     } else {
         bail!("Your Shuttle next project must be a library. Please add `[lib]` to your Cargo.toml file.")
     }
 }
 
-/// Ensure `panic = "abort"` is not set:
-fn check_no_panic(ws: &Workspace) -> anyhow::Result<()> {
-    if let Some(profiles) = ws.profiles() {
-        for profile in profiles.get_all().values() {
-            if profile.panic.as_deref() == Some("abort") {
-                return Err(anyhow!("Your Shuttle project cannot have panics that abort. Please ensure your Cargo.toml does not contain `panic = \"abort\"` for any profiles."));
-            }
+fn is_cdylib(target: &Target) -> bool {
+    target.kind.iter().any(|kind| kind == "cdylib")
+}
+
+async fn compile(
+    packages: Vec<&Package>,
+    release_mode: bool,
+    wasm: bool,
+    project_path: PathBuf,
+) -> anyhow::Result<Vec<BuiltService>> {
+    let manifest_path = project_path.join("Cargo.toml");
+
+    let mut cargo = tokio::process::Command::new("cargo");
+
+    cargo.arg("build").arg("--manifest-path").arg(manifest_path);
+
+    #[cfg(not(debug_assertions))]
+    cargo.arg("-j").arg(4.to_string());
```

Comment:
This should not be added based on if the Shuttle crate is compiled in dev/release mode.
Ideally, the limit should only be imposed in deployment, so I suggest adding a bool `in_deployment` to this function and callers up the chain. The call from cargo-shuttle would use false and the one in deployer true (test cases could use false I guess).

Reply:
Indeed, I think we did this before adding support for local runs with release profile. For context, the reason we set this for deploys is because cargo will think that the deployer has full to all of our VMs cores, even though it is limited by docker. Additionally using a lot of threads for compiling will increase peak memory usage for builds.